src/spend/secret.rs

```diff
@@ -0,0 +1,143 @@
+use super::public::PublicKey;
+use super::stealth::StealthAddress;
+
+use crate::sponge;
+use crate::{JubJubScalar, ViewKey};
+
+use dusk_jubjub::GENERATOR_EXTENDED;
+
+use std::fmt;
+
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+use rand::{CryptoRng, RngCore};
+use sha2::{Digest, Sha256};
+
+/// Secret pair of `a` and `b`
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SecretKey {
+    a: JubJubScalar,
+    b: JubJubScalar,
+}
+
+impl Default for SecretKey {
+    fn default() -> Self {
+        SecretKey {
+            a: JubJubScalar::random(&mut rand::thread_rng()),
+            b: JubJubScalar::random(&mut rand::thread_rng()),
+        }
+    }
+}
+
+impl SecretKey {
+    /// This method is used to construct a new `SecretKey` from the given secret
+    /// pair of `a` and `b`.
+    pub fn new(a: JubJubScalar, b: JubJubScalar) -> Self {
+        Self { a, b }
+    }
+
+    /// Gets `a`
+    pub fn a(&self) -> &JubJubScalar {
+        &self.a
+    }
+
+    /// Gets `b`
+    pub fn b(&self) -> &JubJubScalar {
+        &self.b
+    }
+
+    /// Generate a `sk_r = H(a · R) + b`
+    pub fn sk_r(&self, sa: &StealthAddress) -> JubJubScalar {
+        let aR = sa.R() * self.a;
+        let aR = sponge::hash(&aR);
+        aR + self.b
+    }
+
+    /// Deterministically create a new [`SecretKey`] from a random number
+    /// generator
+    pub fn from_rng<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
```

Comment:
The convention is to call it `random` but it's just an idea. So this is also fine if you want to keep it.

Reply:
I don't mind, I just left the original name, I can replace with `random`, sounds good to me.

Reply:
Whatever you prefer. All of the other repos we use have `random` so probably makes sense to stick with it and be consistent.